File: 3_point-to-point_comm/3.0_comm_p2p.c
#include <stdio.h>
#include <stdlib.h>
#include <FIXME>

void main(int argc, char * argv[]) {
     int err;
     int nprocs, my_rank;
     FIXME;
     MPI_Comm_FIXME(MPI_COMM_WORLD, &nprocs);
     MPI_Comm_FIXME(MPI_COMM_WORLD, &my_rank);

     if (nprocs != 2 && my_rank == 0) {
       printf("ERROR: this code should be run with exactly 2 processes");
       exit(1);
     }       

     MPI_Barrier(MPI_COMM_WORLD);
     
     FIXME *status;
     int number;
     if (my_rank == FIXME) { 
       number = 0;
       printf("Before sending: my number is %d and I am proc number %d \n", FIXME, FIXME);
       MPI_Send(FIXME);
     }
     else if (my_rank == FIXME)  {
       number = 1;
       printf("Before receiving: my number is %d and I am proc number %d \n", number, my_rank);
       MPI_Recv(FIXME);
     }

     printf("My number is %d and I am proc number %d \n", number, my_rank);

     FIXME;
}

File: 3_point-to-point_comm/3.0_comm_p2p.f90
program my_prog
   use FIXME
   implicit none
   integer :: ierr, nprocs, my_rank, a
   integer status(FIXME)
   FIXME
   call mpi_FIXME(mpi_comm_world, nprocs, ierr)
   call mpi_FIXME(mpi_comm_world, my_rank, ierr)

   if (nprocs .NE. 2 .AND. my_rank .EQ. 0) then
      write (*,*) "ERROR: this fortran code should be run with exactly two tasks"
      call exit(1)
   end if
   
   IF( my_rank .EQ. FIXME ) THEN
      a = 10
      CALL MPI_SEND(FIXME)
   ELSE IF( my_rank .EQ. FIXME ) THEN
      a = 3
      CALL MPI_RECV(FIXME)
   END IF
   
   write(*,*) "Hello I am rank number ", my_rank, " and my value of a is ", a
   FIXME
end program my_prog
